src/common/mfu_io.c

```diff
@@ -57,7 +255,28 @@ int mfu_lchown(const char* path, uid_t owner, gid_t group)
     return rc;
 }
 
-/* calls chmod, and retries a few times if we get EIO or EINTR */
+int daos_chmod(const char *path, mode_t mode, mfu_file_t* mfu_file)
+{
+#ifdef DAOS_SUPPORT
+    int rc;
+    dfs_obj_t *parent = NULL;
+    char *name = NULL, *dir_name = NULL;
+    parse_filename(path, &name, &dir_name);
+    assert(dir_name);
+    rc = dfs_lookup(mfu_file->dfs, dir_name, O_RDWR, &parent, NULL, NULL);
+    if (parent == NULL) {
+        fprintf(stderr, "dfs_lookup %s failed \n", dir_name);
+        errno = ENOENT;
+    }
+    rc = dfs_chmod(mfu_file->dfs, parent, name, mode);
+    if (rc) {
+        fprintf(stderr, "dfs_chmod %s failed (%d)\n", name, rc);
+        errno = rc;
+    }
+    return rc;
```

Comment:
```suggestion
    char* name = NULL;
    char* dir_name = NULL;
    parse_filename(path, &name, &dir_name);
    assert(dir_name);

    dfs_obj_t* parent = NULL;
    int rc = dfs_lookup(mfu_file->dfs, dir_name, O_RDWR, &parent, NULL, NULL);
    if (parent != NULL) {
        rc = dfs_chmod(mfu_file->dfs, parent, name, mode);
        if (rc) {
            MFU_LOG(MFU_LOG_ERR, "dfs_chmod %s failed (%d %s)", name, rc, strerror(rc));
            errno = rc;
            rc = -1;
        }
    } else {
        MFU_LOG(MFU_LOG_ERR, "dfs_lookup %s failed", dir_name);
        errno = ENOENT;
        rc = -1;
    }
    return rc;
```

Reply:
What do you think of this?  Hopefully, I didn't mess something up there.  A few changes:

- avoid calling dfs_chmod if parent == NULL
- changed daos_chmod to return -1 on any error
- replaced fprintf with MFU_LOG
- a few style changes